### PR TITLE
Update TextStyle API Docs

### LINKED ISSUE #79292: document platform font family defaults
platform defaults should be documented, what font family will be used by default in different environments

### PR DESCRIPTION
**Pull request by omchiii**
Closes  [flutter/issues/79292](https://github.com/flutter/flutter/issues/79292)

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by Saifallak** [comment]
isn't the default font for android Roboto?

File: packages/flutter/lib/src/painting/text_style.dart [hunk 1]
```diff
@@ -426,6 +426,13 @@ const double _kDefaultFontSize = 14.0;
 ///
 /// ### Inconsistent platform fonts
 ///
+/// By default, fonts differ depending on the platform. 
+///
+///  * Default font-family for `Android`,`Fuchsia` and `Linux` is `Segoe UI`.
```
Comment (Saifallak) [inline packages/flutter/lib/src/painting/text_style.dart:431]: isn't the default font for android Roboto?

Comment (omchiii) [reply packages/flutter/lib/src/painting/text_style.dart:431]: Oh, my bad.

Comment (omchiii) [reply packages/flutter/lib/src/painting/text_style.dart]: Yes. My bad

**Review by darrenaustin — CHANGES_REQUESTED** [review]
Thanks for the contribution. There are a couple of failing analyzer tests (looks like extra space at the end of some lines) that will need to be fixed. 

Also, you will need to get a test exemption from Hixie in the discord chat as described above.

Otherwise it looks good.

File: packages/flutter/lib/src/painting/text_style.dart [hunk 2]
```diff
@@ -426,6 +426,13 @@ const double _kDefaultFontSize = 14.0;
 ///
 /// ### Inconsistent platform fonts
 ///
+/// By default, fonts differ depending on the platform.
+///
+///  * Default font-family for `Android`,`Fuchsia` and `Linux` is `Roboto`.
+///  * Default font-family for `iOS` is `.SF UI Display`/`.SF UI Text`.
+///  * Default font-family for `MacOS` is `.AppleSystemUIFont`.
+///  * Default font-family for `Windows` is `Segoe UI`.
```
Comment (Hixie) [inline packages/flutter/lib/src/painting/text_style.dart:434]: For optimal grammar correctness, `The default font-family for...`

**Comment by Hixie** [comment]
test-exempt: only affects comments

**Comment by Hixie** [comment]
It would be great to leave a comment (not part of the dartdocs, just a literal comment) that says where this is implemented, so that the next person who wants to update this can easily look up the defaults.

**Comment by nt4f04uNd** [comment]
I think this doesn't fully resolve #83392

**Comment by omchiii** [comment]
> I think this doesn't fully resolve #83392

You are right. I linked the correct url but the reference # was wrong. It should be good now.

**Comment by nt4f04uNd** [comment]
Yep, I actually meant the correct one (#79292) =)

File: packages/flutter/lib/src/painting/text_style.dart (hunk 2 above)
Comment (omchiii) [reply packages/flutter/lib/src/painting/text_style.dart]: done.

**Comment by omchiii** [comment]
> It would be great to leave a comment (not part of the dartdocs, just a literal comment) that says where this is implemented, so that the next person who wants to update this can easily look up the defaults.

Is it okay now?

File: packages/flutter/lib/src/painting/text_style.dart [hunk 3]
```diff
@@ -426,6 +426,14 @@ const double _kDefaultFontSize = 14.0;
 ///
 /// ### Inconsistent platform fonts
 ///
+/// By default, fonts differ depending on the platform.
+///
+///  * The default font-family for `Android`,`Fuchsia` and `Linux` is `Roboto`.
+///  * The default font-family for `iOS` is `.SF UI Display`/`.SF UI Text`.
+///  * The default font-family for `MacOS` is `.AppleSystemUIFont`.
+///  * The default font-family for `Windows` is `Segoe UI`.
+// The implementation of differing fonts(depending on the platform) can be found in /packages/flutter/lib/src/material/typography.dart
```
Comment (darrenaustin) [inline packages/flutter/lib/src/painting/text_style.dart:435]: I would add a separator line // between the two comment blocks. I don't think we need the '(depending on the platform)' part.  How about something like:

```dart
/// * The default font-family for `Windows` is `Segoe UI`.
//
// The implementation of these defaults can be found in:
// /packages/flutter/lib/src/material/typography.dart'
```

**Review by darrenaustin — COMMENTED** [review]
Just some minor formatting on the new comment and I think we are good to go.

File: packages/flutter/lib/src/painting/text_style.dart [hunk 4]
```diff
@@ -426,6 +426,15 @@ const double _kDefaultFontSize = 14.0;
...
 /// ### Inconsistent platform fonts
 ///
+/// By default, fonts differ depending on the platform.
+///
+///  * The default font-family for `Android`,`Fuchsia` and `Linux` is `Roboto`.
+///  * The default font-family for `iOS` is `.SF UI Display`/`.SF UI Text`.
+///  * The default font-family for `MacOS` is `.AppleSystemUIFont`.
+///  * The default font-family for `Windows` is `Segoe UI`.
+//
+// The implementation of these defaults can be found in /packages/flutter/lib/src/material/typography.dart
```
Comment (darrenaustin) [inline packages/flutter/lib/src/painting/text_style.dart:436]: Sorry to be picky, but can you break this up into two lines:

```dart
// The implementation of these defaults can be found in:
// /packages/flutter/lib/src/material/typography.dart
```

As that makes it a little easier to read on narrow displays.

File: packages/flutter/lib/src/painting/text_style.dart (hunk 3 above)
Comment (omchiii) [reply packages/flutter/lib/src/painting/text_style.dart]: @darrenaustin done.

**Review by darrenaustin — COMMENTED** [review]
Small nit.

File: packages/flutter/lib/src/painting/text_style.dart (hunk 4 above)
Comment (omchiii) [reply packages/flutter/lib/src/painting/text_style.dart]: @darrenaustin done.

**Review by darrenaustin — APPROVED** [review]
flutter-lgtm

Congrats on your first contribution to the Flutter repo!

**Pull request merged**